extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/security/HttpSecurityRecorder.java

```diff
@@ -212,9 +217,17 @@ public void accept(Throwable throwable) {
                 proceed(throwable);
             } else if (throwable instanceof AuthenticationRedirectException redirectEx) {
                 event.response().setStatusCode(redirectEx.getCode());
-                event.response().headers().set(HttpHeaders.LOCATION, redirectEx.getRedirectUri());
                 event.response().headers().set(HttpHeaders.CACHE_CONTROL, "no-store");
                 event.response().headers().set("Pragma", "no-cache");
+
+                if (redirectEx.getCode() == 200) {
+                    // The target URL is embedded in the auto-submitted form post payload
+                    log.debugf("Form post redirect to %s", redirectEx.getRedirectUri());
+                    event.response().write(redirectEx.getRedirectUri());
```

Comment:
I wonder if you shouldn't do something like `event.response().write("").onComplete(res -> proceed(redirectEx));`. Sadly https://github.com/eclipse-vertx/vert.x/blob/420d4cc543973cd42c4c3af4c0789495c7d77d64/vertx-core/src/main/java/io/vertx/core/http/HttpServerResponse.java#L221 is not super descriptive and I am not able to tell from the implementation if it is necessary :shrug: .

Reply:
Yeah, I was thinking about it today, but we have `end` calls everywhere with these calls returning `Future<Void>` which we don't handle.

Reply:
@michalvavrik But I think it is safer to use your option,  I'll do it